### PR TITLE
Enable nullability in some ControlDesigner members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some ControlDesigner members.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11425)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 1]
```diff
@@ -1624,15 +1624,14 @@ protected virtual void OnSetCursor()
...
         // Handle shadowed properties
         string[] shadowProps = ["Visible", "Enabled", "AllowDrop", "Location", "Name"];
 
         Attribute[] empty = [];
         for (int i = 0; i < shadowProps.Length; i++)
         {
-            prop = (PropertyDescriptor)properties[shadowProps[i]];
+            PropertyDescriptor? prop = (PropertyDescriptor?)properties[shadowProps[i]];
             if (prop is not null)
             {
```
Comment (LeafShi1) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1636]: Consider using pattern matching here and in the following code
```
 if (properties[shadowProps[i]] is { } prop)
 {
     properties[shadowProps[i]] = TypeDescriptor.CreateProperty(typeof(ControlDesigner), (PropertyDescriptor)prop, empty);
 }
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 2]
```diff
@@ -1453,15 +1451,15 @@ protected virtual void OnMouseDragMove(int x, int y)
...
+        Control? parent = ctl;
+        object? parentDesigner = null;
 
         while (parentDesigner is null && parent is not null)
         {
             parent = parent.Parent;
             if (parent is not null)
             {
-                object d = _host.GetDesigner(parent);
+                object? d = _host?.GetDesigner(parent);
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1462]: Let's go ahead and spell out the variable name
```suggestion
                object? designer = _host?.GetDesigner(parent);
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 3]
```diff
@@ -1482,15 +1480,15 @@ protected virtual void OnMouseEnter()
...
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1491]: Ditto here and changes below as well

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 4]
```diff
@@ -1624,24 +1622,21 @@ protected virtual void OnSetCursor()
...
+                properties[shadowProps[i]] = TypeDescriptor.CreateProperty(typeof(ControlDesigner), prop, []);
             }
         }
 
         // replace this one separately because it is of a different type (DesignerControlCollection) than the
         // original property (ControlCollection)
-        PropertyDescriptor controlsProp = (PropertyDescriptor)properties["Controls"];
+        PropertyDescriptor? controlsProp = (PropertyDescriptor?)properties["Controls"];
 
         if (controlsProp is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1641]:
```suggestion
        if (properties["Controls"] is PropertyDescriptor controlsProp)
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs [hunk 5]
```diff
@@ -1654,7 +1649,7 @@ protected override void PreFilterProperties(IDictionary properties)
                 attrs);
         }
 
-        PropertyDescriptor sizeProp = (PropertyDescriptor)properties["Size"];
+        PropertyDescriptor? sizeProp = (PropertyDescriptor?)properties["Size"];
         if (sizeProp is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ControlDesigner.cs:1653]:
```suggestion
        if (properties["Size"] is PropertyDescriptor sizeProp)
```

**Review by lonitra — APPROVED** [review]

**Pull request merged**